op-deployer/pkg/deployer/pipeline/opchain.go

```diff
@@ -194,6 +139,82 @@ func setPreimageOracleAddress(ctx context.Context, client *ethclient.Client, err
 	errCh <- err
 }
 
+func DeployOPChainGenesisStrategy(env *Env, intent *state.Intent, st *state.State, chainID common.Hash) error {
+	lgr := env.Logger.New("stage", "deploy-opchain", "strategy", "genesis")
+
+	if !shouldDeployOPChain(intent, st, chainID) {
+		lgr.Info("opchain deployment not needed")
+		return nil
+	}
+
+	thisIntent, err := intent.Chain(chainID)
+	if err != nil {
+		return fmt.Errorf("failed to get chain intent: %w", err)
+	}
+
+	input := makeDCI(thisIntent, chainID, st)
+
+	env.L1ScriptHost.ImportState(st.L1StateDump.Data)
+
+	var dco opcm.DeployOPChainOutput
+	lgr.Info("deploying OP chain using local allocs", "id", chainID.Hex())
+	dco, err = opcm.DeployOPChain(
+		env.L1ScriptHost,
+		input,
+	)
+	if err != nil {
+		return fmt.Errorf("error deploying OP chain: %w", err)
+	}
+
+	st.Chains = append(st.Chains, makeChainState(chainID, dco))
```

Comment:
I think you'll want this here too.

```suggestion
	st.Chains = append(st.Chains, makeChainState(chainID, dco))
	err = conditionallySetImplementationAddresses(ctx, env.L1Client, intent, st, dco, st.ImplementationsDeployment.OpcmProxyAddress)
```

Reply:
we don't need that there. the implementation addresses are set in the implementation step. there's also no l1 client in the genesis strategy, since we're building the L1 genesis during the pipeline.